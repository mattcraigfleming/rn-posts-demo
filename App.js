import React from 'react';
import {SafeAreaView, StyleSheet, View, Text} from 'react-native';

const App = () => {
  return (
    // SafeAreaView for iOS version 11 or later https://facebook.github.io/react-native/docs/safeareaview
    <SafeAreaView>
      <View style={styles.body}>
        <Text>Home</Text>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  body: {
    backgroundColor: 'white',
  },
});

export default App;
