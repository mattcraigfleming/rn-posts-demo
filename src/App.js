import React from 'react';
import {Provider} from 'react-redux';
import {PersistGate} from 'redux-persist/integration/react';

import {store, persistor} from './store';
import Navigator from './navigation/Navigator';

const App = () => {
  return (
    <Provider store={store}>
      <Navigator />
    </Provider>
  );
};

export default App;
