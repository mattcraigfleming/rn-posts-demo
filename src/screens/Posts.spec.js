import React from 'react';
import {shallow} from 'enzyme';
import configureStore from 'redux-mock-store';

jest.mock('react-native/Libraries/Lists/FlatList', () => 'FlatList');

// imported as a connected component!
import App from '../App';

const middlewares = []; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

const initialState = {
  posts: [],
};

describe('Testing Posts', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<App />, {
      context: {store: mockStore(initialState)},
    }).dive();
    expect(wrapper.dive()).toMatchSnapshot();
  });
});

//   const posts = [
//     {
//       userId: 1,
//       id: 1,
//       title:
//         'sunt aut facere repellat provident occaecati excepturi optio reprehenderit',
//       body:
//         'quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto',
//     },
//     {
//       userId: 1,
//       id: 2,
//       title: 'qui est esse',
//       body:
//         'est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla',
//     },
//     {
//       userId: 1,
//       id: 3,
//       title: 'ea molestias quasi exercitationem repellat qui ipsa sit aut',
//       body:
//         'et iusto sed quo iure\nvoluptatem occaecati omnis eligendi aut ad\nvoluptatem doloribus vel accusantium quis pariatur\nmolestiae porro eius odio et labore et velit aut',
//     },
//   ];
