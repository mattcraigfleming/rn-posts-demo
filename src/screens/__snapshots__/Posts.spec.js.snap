// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts renders correctly 1`] = `null`;

exports[`Testing Posts renders as expected 1`] = `ShallowWrapper {}`;
